File: src/base/BaseDialog.java
package base;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import com.intellij.ui.components.JBScrollPane;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import javax.swing.border.EmptyBorder;
import java.awt.*;

/**
 * Created by CeH9 on 08.07.2016.
 */
public abstract class BaseDialog extends DialogWrapper {

    protected Project project;
    protected JPanel panel;

    protected static final int PADDING = 10;
    protected final int MIN_WIDTH = 600;
    protected final int MIN_HEIGHT = 520;

    public BaseDialog(@Nullable Project project) {
        super(project);
        this.project = project;
        init();
    }

    public void preShow() {
    }

    public void onOKAction() {
    }

    public void onCancelAction() {
    }

    @Override
    public void show() {
        preShow();
        super.show();

        switch (getExitCode()) {
            case DialogWrapper.OK_EXIT_CODE:
                onOKAction();
                break;
            case DialogWrapper.CANCEL_EXIT_CODE:
                onCancelAction();
                break;
        }
    }

    @Override
    protected JComponent createCenterPanel() {
        panel = new JPanel();
        JBScrollPane scrollPane = new JBScrollPane(panel);
        panel.setBorder(new EmptyBorder(PADDING, PADDING, PADDING, PADDING));
        scrollPane.setMinimumSize(new Dimension(MIN_WIDTH, MIN_HEIGHT));
        return scrollPane;
    }

}

File: src/main/kotlin/ui/base/BaseDialog.kt
package ui.base

import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.DialogWrapper
import java.awt.Dimension
import javax.swing.JComponent
import javax.swing.JPanel

abstract class BaseDialog(
        val project: Project,
        canBeParent: Boolean = true,
        title: String = "Unknown",
        private var width: Int = 600,
        private var height: Int = 400,
        protected val root: JComponent = JPanel()
) : DialogWrapper(project, canBeParent) {

    init {
        super.init()
        this.title = title
    }

    abstract fun initUI()

    override fun getPreferredFocusedComponent() = root

    override fun createCenterPanel(): JComponent? {
        root.minimumSize = Dimension(width, height)
        initUI()

        return root
    }
}